src/gui/instrument/EnvelopeGraph.cpp

```diff
@@ -54,102 +56,208 @@ void EnvelopeGraph::modelChanged()
 
 void EnvelopeGraph::mousePressEvent(QMouseEvent* me)
 {
-	if(me->button() == Qt::LeftButton)
+	if (me->button() == Qt::LeftButton) { toggleAmountModel(); }
+}
+
+void EnvelopeGraph::contextMenuEvent(QContextMenuEvent* event)
+{
+	QMenu menu(this);
+	QMenu* scalingMenu = menu.addMenu(tr("Scaling"));
+	scalingMenu->setToolTipsVisible(true);
+
+	auto switchTo = [&](ScalingMode scaling)
 	{
-		toggleAmountModel();
-	}
+		if (m_scaling != scaling)
+		{
+			m_scaling = scaling;
+			update();
+		}
+	};
+
+	auto addScalingEntry = [scalingMenu, &switchTo, this](const QString& text, const QString& toolTip, ScalingMode scaling)
+	{
+		QAction* action = scalingMenu->addAction(text, [&switchTo, scaling]() { switchTo(scaling); });
```

Comment:
I think we can just use `[&]` instead of passing individual variables and such.

Reply:
Unfortunately not. I have tried this before settling on this solution and it did not catch the `switchTo` lambda and did not work.